actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -633,14 +668,26 @@ def initialize(parent_jar)
       private
         def parse(name, encrypted_message)
           deserialize(name) do |rotate|
-            @encryptor.decrypt_and_verify(encrypted_message, on_rotation: rotate)
+            if cookies_with_metadata?
+              value = @encryptor.decrypt_and_verify(encrypted_message, on_rotation: rotate, purpose: name)
+              if value.nil? && read_cookies_without_metadata?
+                value = @encryptor.decrypt_and_verify(encrypted_message, on_rotation: rotate)
+              end
+              value
+            else
+              @encryptor.decrypt_and_verify(encrypted_message, on_rotation: rotate)
```

Comment:
Calling `@encryptor.decrypt_and_verify(encrypted_message, on_rotation: rotate)` 3 times here is a smell.